daemon/src/maker_cfd.rs

```diff
@@ -471,130 +484,81 @@ where
-        // Since rejection is tied to the cfd state at the moment we can only do this after creating
-        // a cfd.
-        if quantity < current_order.min_quantity || quantity > current_order.max_quantity {
-            tracing::warn!(
-                "Order rejected because quantity {} was out of bounds. It was either <{} or >{}",
-                quantity,
-                current_order.min_quantity,
-                current_order.max_quantity
-            );
+        // 4. Try to get the oracle announcement, if that fails we should exit prior to changing any
+        // state
+        let announcement = self
+            .oracle_actor
+            .send(oracle::GetAnnouncement(cfd.order.oracle_event_id))
+            .await??;
 
-            self.reject_order(taker_id, cfd, conn).await?;
-        }
+        // 5. Start up contract setup actor
+        let this = ctx
+            .address()
+            .expect("actor to be able to give address to itself");
+        let (addr, fut) = setup_maker::Actor::new(
+            (cfd.order, cfd.quantity_usd, self.n_payouts),
+            (self.oracle_pk, announcement),
+            &self.wallet,
+            &self.wallet,
+            (&self.takers, &self.takers, taker_id),
+            &this,
+            (&self.takers, &this),
+        )
+        .create(None)
+        .run();
+
+        disconnected.insert(addr);
+
+        self.tasks.add(fut);
 
         Ok(())
     }
 }
 
 #[xtra_productivity]
-impl<O, M, T, W> Actor<O, M, T, W>
-where
-    Self: xtra::Handler<CfdSetupCompleted>,
-    O: xtra::Handler<oracle::GetAnnouncement>,
-    T: xtra::Handler<maker_inc_connections::TakerMessage>,
-    W: xtra::Handler<wallet::Sign> + xtra::Handler<wallet::BuildPartyParams>,
-{
-    async fn handle_accept_order(
-        &mut self,
-        msg: AcceptOrder,
-        ctx: &mut Context<Self>,
-    ) -> Result<()> {
-        let AcceptOrder { order_id } = msg;
-
-        if let SetupState::Active { .. } = self.setup_state {
-            anyhow::bail!("Already setting up a contract!")
-        }
+impl<O, M, T, W> Actor<O, M, T, W> {
+    async fn handle_accept_order(&mut self, msg: AcceptOrder) -> Result<()> {
+        tracing::debug!("handling accept order from user");
 
-        tracing::debug!(%order_id, "Maker accepts an order" );
+        let AcceptOrder { order_id } = msg;
 
         let mut conn = self.db.acquire().await?;
 
         // 1. Validate if order is still valid
         let mut cfd = load_cfd_by_order_id(order_id, &mut conn).await?;
-        let taker_id = match cfd {
+        match cfd {
             Cfd {
-                state: CfdState::IncomingOrderRequest { taker_id, .. },
+                state: CfdState::IncomingOrderRequest { .. },
                 ..
-            } => taker_id,
+            } => {}
             _ => {
                 anyhow::bail!("Order is in invalid state. Ignoring trying to accept it.")
             }
         };
```

Comment:
I think we can drop this now, the state within the actor is not going to change!